File: packages/java/src/main/java/io/opensms/Sandbox.java
package io.opensms;

import io.opensms.models.Page;
import io.opensms.models.SandboxMessage;

/** Sandbox inspection. Accessed as {@code client.sandbox()}. */
public final class Sandbox {

    private final ApiTransport t;

    Sandbox(ApiTransport t) {
        this.t = t;
    }

    /** @return the first page of sandbox messages. */
    public Page<SandboxMessage> listMessages() {
        return listMessages(new ListParams());
    }

    /**
     * Messages sent with a test key, with their rendered text (including OTP codes).
     *
     * @param params paging.
     * @return one page.
     */
    public Page<SandboxMessage> listMessages(ListParams params) {
        return t.get("/v1/sandbox/messages", params.toQuery(), ApiTransport.pageOf(SandboxMessage.class));
    }
}

File: packages/php/src/Resources/Sandbox.php
<?php

declare(strict_types=1);

namespace Opensms\Resources;

use Opensms\Models;
use Opensms\Page;
use Opensms\Transport;

/**
 * The `sandbox` resource: the rendered text of sandbox sends (including OTP codes).
 * Accessed as `$client->sandbox`.
 */
final class Sandbox
{
    /** @internal */
    public function __construct(private readonly Transport $transport)
    {
    }

    /**
     * @param array<string, mixed> $params `limit`, `cursor`
     */
    public function listMessages(array $params = []): Page
    {
        return Models::page($this->transport->request('GET', '/v1/sandbox/messages', Models::map($params, Models::PAGE_QUERY, [], true)));
    }
}
